internal/db/storage/blob/cloudstorage/cloudstorage.go

```diff
@@ -0,0 +1,80 @@
+package cloudstorage
+
+import (
+	"context"
+	"io"
+
+	iblob "github.com/vdaas/vald/internal/db/storage/blob"
+	"github.com/vdaas/vald/internal/db/storage/blob/cloudstorage/reader"
+	"github.com/vdaas/vald/internal/db/storage/blob/cloudstorage/writer"
+	"github.com/vdaas/vald/internal/errors"
+	"gocloud.dev/blob"
+)
+
+type client struct {
+	urlstr string
+
+	bucket *blob.Bucket
+
+	readerOpts *blob.ReaderOptions
+	writerOpts *blob.WriterOptions
+}
+
+// New returns blob.Bucket implementation.
+func New(opts ...Option) (iblob.Bucket, error) {
+	c := new(client)
+
+	for _, opt := range append(defaultOpts, opts...) {
+		if err := opt(c); err != nil {
+			return nil, err
+		}
+	}
+	return c, nil
+}
+
+func (c *client) Open(ctx context.Context) (err error) {
+	c.bucket, err = blob.OpenBucket(ctx, c.urlstr)
+	if err != nil {
+		return err
+	}
+	return
+}
+func (c *client) Close() error {
+	if c.bucket == nil {
+		return errors.ErrBucketNotOpened
+	}
+	return c.bucket.Close()
+}
+
+func (c *client) Reader(ctx context.Context, key string) (io.ReadCloser, error) {
+	if c.bucket == nil {
+		return nil, errors.ErrBucketNotOpened
+	}
+
+	r, err := reader.New(
+		reader.WithKey(key),
+		reader.WithBucket(c.bucket),
+		reader.WithReaderOptions(c.readerOpts),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, r.Open(ctx)
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg?style=flat-square)
i think there's no need to create `reader` package.
```suggestion
	return c.bucket.NewReader(ctx, key, c.readerOpts)
```

This is just my opinion. If you'd like to do that, it is okay.

Reply:
I think your opinion is good. 

However, it is difficult to test with the current structure, so can I make c.buckt a private interface?
the following is `example`.
```go
type bucket interface {
     NewReader(ctx context.Context, key string, opts *ReaderOptions) (*Reader, error)
     NewWriter(ctx context.Context, key string, opts *WriterOptions) (*Writer, err error)
     Close()
}

```

Reply:
It is okay to add a new interface, but i think it is much easier to use the memblob package for test use.
https://godoc.org/gocloud.dev/blob/memblob

Reply:
Thank you for your suggestion.
I did not know this pacakge. 
Thank you and I use it.